### ISSUE TITLE
[v2-rc9] $inc for nested key doesn't work

### ISSUE THREAD

**Issue by talha-asad** [jagi/meteor-astronomy#360]
The below code was working fine before I upgraded to rc9, I had a cloned copy from rc8. Any idea why it stopped working in the latest rc?

``` js
SomeCollection.update({_id: this.userId}, {$inc: {'stats.some_count': 1}});
```

**Comment by lukejagodzinski** [comment]
Hmm strange. I haven't been playing with this at all. Maybe you have changed something in your code? If not then please create simple reproduction.

**Comment by talha-asad** [comment]
If you try incrementing any nested key using the Astro Class `.update` it returns 1 but doesn't actually make any update in the db. It works fine if the same query is performed using `Meteor.collection`

Could it have something to do with the minimongo available in 1.3.2.4?

So basically this will work:

``` js
SomeCollection.update({_id: this.userId}, {$inc: {'some_count': 1}});
```

but this won't work:

``` js
SomeCollection.update({_id: this.userId}, {$inc: {'stats.some_count': 1}});
```

**Comment by lukejagodzinski** [comment]
Hmm I don't know if it has anything to do with Minimongo but I will have to check it. I know there was a problem with `@inc` operator but I've fixed it but using some internal Minimongo methods. So maybe in the update they removed something. I will try to check it today or tomorrow.

**Comment by lukejagodzinski** [comment]
In deed it's a bug, but I'm 100% sure that it used to work. I will try fixing it

**Comment by lukejagodzinski** [comment]
Ok bug fixed, 2.0 is very very close :)
